org.adempiere.base/src/org/compiere/util/Env.java

```diff
@@ -2101,4 +2103,26 @@ public static int getZoomWindowID(int AD_Table_ID, int Record_ID, int windowNo)
 		getCtx().put(LANGUAGE, Language.getBaseAD_Language());
 	}   //  static
 
+
+	public static void setPredefinedVariables(Properties ctx, int windowNo, String predefinedVariables) {
+		if (predefinedVariables != null) {
+			String[] lines = predefinedVariables.split("\n");

```

Comment:
this is split by newline ? I think comma or semi colon separated is easier for entry (would be nice if user can use either one of the split character).

Reply:
I thought about this but is very easy that comma or semi-colon is added inside a context variable, and to allow that we would need a parser that checks for quotation, and then a parser that checks for escape character (\\)
So, in principle with newline approach we just cannot add newlines within these context variables